packages/core/src/use-router-query.ts

```diff
@@ -0,0 +1,8 @@
+import {useRouter} from 'next/router'
+import {parse} from 'url'
```

Comment:
Need to add `url` as a dependency of the `blitzjs/core` package

Reply:
`url` dependency is added. I've also rewritten the implementation of `useRouterParams` as you described, replaced `router.query` in the store example with new hooks and added integration test that tests if `/admin/products?order=asc` changes the order of products in DOM

> Lastly, before merging this we should resolve our discussion in #530 about number parsing. Maybe we can have `useParamsStr`, `useParamsInt`, and `useRouterQuery`

Let's discuss it in the issue.